### PR TITLE
Update readme and Cargo.toml

### PR DESCRIPTION
**Pull request by sandreim**
## Reason for This PR

Update readme and Cargo.toml files to reflect the experimental status of the crate.

## Description of Changes

Provided in commit messages.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] The reason for this PR is clearly provided (issue no. or explanation).
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] Any newly added `unsafe` code is properly documented.
- [ ] Any user-facing changes are mentioned in `CHANGELOG.md`.

### CODE REVIEW

**Review by serban300 — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -1,2 +1,7 @@
 # versionize_derive
+## Important note
+
+This crate is currenlty used for cross-version serialization with the [Firecracker snapshot-restore dev preview](https://github.com/firecracker-microvm/firecracker/tree/v0.23.0), but has not been tested for other use cases. It should be considered **experimental software** outside the Firecracker context. It’s likely that this crate will see both interface and implementation changes in the future.
```
Comment (nchong-at-aws) [inline README.md]: currently

Comment (sandreim) [reply README.md]: Thanks Nathan!

**Comment by sandreim** [comment]
Updated PR, please take another look.

**Review by serban300 — DISMISSED** [review]

File: README.md [hunk 2]
```diff
@@ -1,2 +1,7 @@
 # versionize_derive
+## Important note
```
Comment (raduweiss) [inline README.md:3]:
```suggestion

## Important note
```

I think markdown formatting requires 1 free line between two heading/paragraph elements.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,7 @@
...
+
+This crate is currently used for cross-version serialization with the [Firecracker snapshot-restore dev preview](https://github.com/firecracker-microvm/firecracker/tree/v0.23.0), but has not been tested for other use cases. It should be considered **experimental software** outside the Firecracker context. It’s likely that this crate will see both interface and implementation changes in the future.
+
+## Description
```
Comment (raduweiss) [inline README.md:7]:
```suggestion
## Description

```

I think markdown formatting requires 1 free line between two heading/paragraph elements.

**Review by raduweiss — DISMISSED** [review]
Two white-space comments, otherwise good to go.

**Review by raduweiss — APPROVED** [review]

**Review by serban300 — APPROVED** [review]

**Pull request merged**